p2p/src/sync/peer_v2/txn_mgr.rs

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) 2021-2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeSet;
+
+use tokio::sync::mpsc::{Receiver, UnboundedReceiver, UnboundedSender};
+
+use common::{
+    chain::{ChainConfig, Transaction},
+    primitives::{Id, Idable},
+    time_getter::TimeGetter,
+};
+use logging::log;
+use mempool::MempoolHandle;
+use utils::const_value::ConstValue;
+use utils::sync::Arc;
+
+use crate::{
+    config::P2pConfig,
+    error::{P2pError, ProtocolError},
+    message::{TransactionResponse, TxnSyncMessage},
+    net::{
+        types::services::{Service, Services},
+        NetworkingService,
+    },
+    sync::{
+        chainstate_handle::ChainstateHandle,
+        peer_common::{handle_message_processing_result, KnownTransactions},
+        LocalEvent,
+    },
+    types::peer_id::PeerId,
+    MessagingService, PeerManagerEvent, Result,
+};
+
+// TODO: Take into account the chain work when syncing.
+/// Transaction sync manager.
+///
+/// Syncing logic runs in a separate task for each peer.
+pub struct PeerTxnSyncManager<T: NetworkingService> {
+    id: ConstValue<PeerId>,
+    _chain_config: Arc<ChainConfig>,
+    p2p_config: Arc<P2pConfig>,
+    common_services: Services,
+    chainstate_handle: ChainstateHandle,
+    mempool_handle: MempoolHandle,
+    peer_manager_sender: UnboundedSender<PeerManagerEvent>,
+    messaging_handle: T::MessagingHandle,
+    sync_msg_rx: Receiver<TxnSyncMessage>,
+    local_event_rx: UnboundedReceiver<LocalEvent>,
+    _time_getter: TimeGetter,
```

Comment:
Should the unused members (chain config, time getter) be just removed for now? Can be added again quite easily if needed.